### ISSUE TITLE
Crash after upgrade to 4.17.5

### ISSUE THREAD

**Issue by tuxayo** [signalapp/Signal-Android#7467]
I have:
- [X] searched open and closed issues for duplicates
- [X] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
I just upgraded to v4.16.9 from the PlayStore. And have the same symptom as #7386.
But v4.16.9 should have fix it so it's not the very same issue.
I didn't remember a message about any kind of database update.

### Workaround
Downloaded and installed Signal 4.15.5 from the website: https://signal.org/android/apk/
Download link here if someone else need this same version and the site is updated: https://updates.signal.org/android/Signal-website-release-4.15.5.apk

I'm immensely grateful to @moxie0 for making such a robust app in this precise downgrading scenario. (well I might be just lucky) I was terrified that the new version might have actually done a change in the DB which would have prevented the old version to work. So the downgrade worked.

Now I learned that I should do a full backup of some critical apps like Signal before upgrading. I'm not sure if oandbackup is actually enough to restore an APK **and** the data due to https://github.com/signalapp/Signal-Android/issues/7457#issuecomment-370167427
So according to that, uninstalling the will prevent the backup to work. So plain text backup should still work right?

#### The workaround isn't actually enough
What would be a way to upgrade to the latest version? (I retried the process and it didn't work) What additional info can I provide?

### Additional info
This is a more than 3 years old install. I'm pretty sure I used oandbackup (which should raise the same issues as Titanium Backup) at some point to troubleshot, test or work around an issue. (including a uninstallation of the app and a restore) I think it happened something like 2 years ago so I don't know if this has consequences.

### Device info
**Device:** Wiko Cink Peax 2
**Android version:** 4.1.2
**Signal version:** 4.16.9

### Link to debug log
When opening the 4.16.9 and seeing no messages.
https://framabin.org/?5aa58fc8356f14e6#YWa9TfMmxzgPK8WaOyIj94N53AX1frs935ntMS6WDvA=

**Comment by moxie0** [comment]
Could you:

1. Install this cleaner APK *over* your existing install (do not uninstall first):
https://mega.nz/#!80sBzI6S!r0eMF8TXLElRBnlXUb608GRn8JcbfZXEoutoJ4wY46M

2. Open the "app"

3. Install 4.16.0 *over* that (do not uninstall first):
https://mega.nz/#!Yk1HQIoa!MzZDZ7cZo1gzTevrLTWTjvVqNGKl66jkGy9nBqrNrsI

3. Run adb logcat

4. Open Signal

5. Post the logcat output here

**Comment by tuxayo** [comment]
It worked, thanks a lot! :D

Here are the logs:
https://framabin.org/?e6267032d2559682#v8bfdrAYt7Hf1QQRgtDXR2I6Y13Omrbsx71/aHZVrmM=

**Comment by tuxayo** [comment]
A lot of crashes making the access to most conversation impossible. Will post the logs soon. (in this issue or a new one?)

**Comment by tuxayo** [comment]
Just upgraded from 4.15.5 to 4.17.5 and I have the same crashes.

Here is the logcat of the whole system.
The actions were:
- airplane mode and stopped the wifi to have less noise in the logs
- open the app
- unlock with passphrase
- scroll the list a bit (until a specific conversation)
- crash

Two logs
https://framabin.org/?0714936dcc5d7750#Ge2vBIGtn5yNHUvGy4dSJ/rcGr7W5aWS24LsZfhEM7c=
Second were I forgot to cut the wifi
https://framabin.org/?0042cb2adba43b03#bG5zcPXhRkg2TzhK2O6q7TtX1Z/Q7FL1yD5ZDsCA2t0=

With the 4.17.5 I only see messages before the 18 march. I think that the last message of the conversion triggers the crash is a picture. So it might be when trying to show thumbnail.

**Comment by tuxayo** [comment]
Is there anything else I can do?

**Comment by moxie0** [comment]
It's hard for me to follow what's going on here. It looks like there was an old issue that was resolved, and then has been reopened and renamed for a different issue? Can you please open a new issue for a new issue, it makes things hard to follow otherwise. Thanks!
